group-coordinator/src/main/java/org/apache/kafka/coordinator/group/runtime/CoordinatorRuntime.java

```diff
@@ -528,8 +528,8 @@ private void transitionTo(
                 case LOADING:
                     state = CoordinatorState.LOADING;
                     snapshotRegistry = new SnapshotRegistry(logContext);
-                    lastWrittenOffset = 0L;
-                    lastCommittedOffset = 0L;
+                    updateLastWrittenOffset(0);
+                    updateLastCommittedOffset(0);
```

Comment:
I think that this won't always work. For instance, think about the case where we go from Failed to Loading. The reset won't work because `0` is not larger than the current values. It may be better to just keep what was there and to add the first snapshot creating there.